src/plugins/discover/public/migrate_state.ts

```diff
@@ -133,7 +133,9 @@ export function migrateUrlState(oldPath: string, newPath = '/'): string {
           indexPattern: index,
         },
       };
+      const _g = getStateFromOsdUrl<any>('_g', oldPath);
```

Comment:
Is this path only entered if the url uses url fragments? I ask because it looks like `parseUrlHash` would just parse the url hash, so if it wasn't using fragments, you'd potentially override a validly set `_g` query parameter.

Also, does your comment about ftr tests address the lack of test coverage here?

Reply:
My current understanding is that the legacy urls are all like http://localhost:5601/app/discover#/, there should be no issue with parsing and handling _g.

Reply:
Will add the unit test.